### DISCUSSION TITLE
supabase function works on sql query but not on rpc??

### DISCUSSION THREAD

**Discussion by MohammedHafiz** [supabase/supabase#5913] [Questions] [answered]
Supabase function works on sql query but does not work on rpc??

Below is how I create my function :-

```sql
create or replace function total_amount_completed_orders()
returns int
language sql
as $$
  SELECT SUM (final_amount) AS total
  FROM completed_orders;
$$;
```

select total_amount_created_orders() returns me the correct answer on sql query.

Below is how I'm making my rpc call :-

```js
const { data, error } = supabase
  .rpc('total_amount_completed_orders')
```

Below is the RLS policy for that table:-

```sql
((uid() = user_id) OR (email() = 'myadminemail@email.com'::text))
```

Currently logged in through my admin email while attempting to make the rpc call.

**Comment by GaryAustin1** [comment, marked as answer]
No errors?  Maybe not signed in when you make the call.  You can temporarily add "auth.role()='anon' OR" to your RLS to check that quickly.

Edit:  It looks like your post changed.  Your email was `'myadminemail@email.com'::text` earlier.   The new format will not work it just needs to be the base email string.

**Reply by MohammedHafiz** [reply]
Hi @GaryAustin1,
Both data and error are undefined for some reason. I am logged in and I can query the same table just fine with my current RLS. I'm not sure why somebody else edited what I had initially added cause clearly it doesn't work as well.

**Reply by GaryAustin1** [reply]
You need await before supabase in your call.

**Reply by MohammedHafiz** [reply]
oh.. thank you @GaryAustin1 it worked now
